google/auth/_refresh_worker.py

```diff
@@ -62,101 +52,53 @@ def start_refresh(self, cred, request):
                 "Unable to start refresh. cred and request must be valid and instantiated objects."
             )
 
-        # This test case is covered by the unit tests but sometimes the cover
-        # check can flake due to the schdule.
-        #
-        # Specifially this test is covered by test_refresh_dead_worker
-        if not self._refresh_queue.empty():  # pragma: NO COVER
-            if self._need_worker():
-                self._spawn_worker()
-            return
+        with self._lock:
+            if self._worker is not None and self._worker._error_info is not None:
+                raise e.RefreshError(
+                    f"Could not start a background refresh. The background refresh previously failed with {self._worker._error_info}."
+                ) from self._worker._error_info
 
-        try:
-            self._refresh_queue.put_nowait((cred, request))
-        except queue.Full:
-            return
-
-        # This test case is covered by the unit tests but sometimes the cover
-        # check can flake due to the schdule.
-        if self._need_worker():  # pragma: NO COVER
-            self._spawn_worker()
-
-    def error_queue_full(self):
-        """
-      True if the refresh worker error queue is full. False if it is not yet full.
-
-      Returns:
-        bool
-      """
-        return self._error_queue.full()
-
-    def flush_error_queue(self):
-        """
-      Drop all errors in the error queue.
-      """
-        try:
-            while not self._error_queue.empty():
-                _ = self._error_queue.get_nowait()
-        # This condition is unlikely but there is a possibility that an
-        # error gets queued between the empty and get calls
-        except queue.Empty:  # pragma: NO COVER
-            pass
+            if self._need_worker():  # pragma: NO COVER
```

Comment:
nit: within a mutex, it's usually better to be explicit about accessing data and not call out to methods.
* this helps restrict the scope of the mutex
* this ensures the methods aren't called outside of a mutex

These methods are simple enough (and similar to the check above) so I'd just inline them.

Reply:
Makes sense, moved these to be inlined